crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -1284,12 +1673,63 @@ fn retain_names_from_declared_idents(child_names: &mut Vec<Name>, current_declar
     *child_names = retained_names;
 }
 
-fn gen_ident(cnt: &mut u32) -> JsWord {
+fn gen_action_ident(cnt: &mut u32) -> JsWord {
     let id: JsWord = format!("$$ACTION_{cnt}").into();
     *cnt += 1;
     id
 }
 
+fn gen_cache_ident(cnt: &mut u32) -> JsWord {
+    let id: JsWord = format!("$$CACHE_{cnt}").into();
+    *cnt += 1;
+    id
+}
+
+fn wrap_cache_expr(expr: Expr, name: &str, id: &str) -> Expr {
```

Comment:
```suggestion
fn wrap_cache_expr(expr: Box<Expr>, name: &str, id: &str) -> Box<Expr> {
```

This is for performance and consistency. `memmove` is much more costly than intuitive estimation.